### DISCUSSION TITLE
batch writes possible?

### DISCUSSION THREAD

**Discussion by zidkim** [supabase/supabase#7636] [Questions] [answered]
Hello Team & Community,

Couldn't find this in the documentation.
Can we do batch writes in supabase.js? 

For example, could we:

```
const batch = supabase.batch();

batch.from("table_one").update({...});
batch.from("table_two").update({...});
batch.from("table_three").update({...});

await batch();
```

The goal is to fail all transactions if one or more transaction fails.

**Comment by GaryAustin1** [comment, marked as answer]
You would have to pass in all data to an rpc function and make your inserts.  The entire function is treated as a transaction so will roll back if any insert fails.

**Reply by zidkim** [reply]
Gotcha, makes sense, thanks!

**Reply by vidicunt** [reply]
Can you give more details please?
